src/components/Common/SearchInput.tsx

```diff
@@ -79,20 +77,83 @@ const KeyboardShortcutHint = ({ open }: { open: boolean }) => {
     </div>
   );
 };
-
-const SearchByMultipleFields: React.FC<SearchByMultipleFieldsProps> = ({
-  id,
+const SearchInputFieldRenderer = ({
+  selectedOption,
+  searchValue,
+  setSearchValue,
+  inputRef,
+  inputClassName,
+  autoFocus,
+  isSingleOption,
+  open,
+  ...prop
+}: {
+  selectedOption: SearchOption;
+  searchValue: string;
+  setSearchValue: (value: string) => void;
+  inputRef?: React.RefObject<HTMLInputElement | null>;
+  inputClassName?: string;
+  autoFocus?: boolean;
+  isSingleOption: boolean;
+  open: boolean;
+}) => {
+  switch (selectedOption.type) {
+    case "phone":
+      return (
+        <div className="relative">
+          <PhoneInput
+            name={selectedOption.key}
+            placeholder={selectedOption.placeholder}
+            value={searchValue}
+            onChange={(value) => setSearchValue(value || "")}
+            className={inputClassName}
+            autoFocus={autoFocus}
+            ref={inputRef}
+            {...prop}
+          />
+          {!isSingleOption && <KeyboardShortcutHint open={open} />}
+        </div>
+      );
+    default:
+      return (
+        <div className="relative">
+          <Input
+            type="text"
+            placeholder={selectedOption.placeholder}
+            ref={inputRef as React.RefObject<HTMLInputElement>}
+            value={searchValue}
+            onChange={(event) => setSearchValue(event.target.value)}
+            className={cn(
+              !isSingleOption &&
+                "grow border-none shadow-none focus-visible:ring-0",
+              inputClassName,
+            )}
+            {...prop}
+          />
+          {!isSingleOption && <KeyboardShortcutHint open={open} />}
+        </div>
+      );
+  }
+};
+export default function SearchInput({
   options,
   onSearch,
-  initialOptionIndex,
   className,
   inputClassName,
   buttonClassName,
-  clearSearch,
   onFieldChange,
   enableOptionButtons = true,
   autoFocus = false,
-}) => {
+  ...props
+}: SearchInputProps) {
+  const initialOptionIndex = useMemo(
+    () =>
+      Math.max(
+        options.findIndex((option) => option.value !== ""),
+        0,
+      ),
+    [options],
+  );
```

Comment:
using `useMemop` for the tiniest things will actually cause performance issues instead of making it better. the overhead brought in by `useMemo` is bad than this.

i don't see any reason for using `useMemo` for this tiny thing